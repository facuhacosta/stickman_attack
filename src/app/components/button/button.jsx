import React from 'react'
import style from './button.module.scss'

export function Button ({ onClick, text, primary, secondary, ternary, animationDelay, type }) {
  const buttonClasses = `${style.button}${primary ? ` ${style.primary}` : ''}${secondary ? ` ${style.secondary}` : ''}${ternary ? ` ${style.ternary}` : ''}`

  return (
    <div className={style.container}>
      <button className={buttonClasses} style={{ animationDelay }} type={type} onClick={onClick}>{text}</button>
    </div>
  )
}
