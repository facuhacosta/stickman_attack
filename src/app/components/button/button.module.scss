.button {
  background: radial-gradient(ellipse,
    rgb(150, 150, 150) 0%,
    rgb(124, 124, 124) 50%,
    rgb(100, 100, 100) 100%);
  background-color: transparent;
  position: relative;
  overflow: hidden;
}

.secondary {
  width: 100px;
  height: 100px;
}

.ternary {
  width: 100px;
  height: 50px;
  font-size: large;
  font-weight: 600;
  border: solid 3px rgb(90, 90, 90);
  border-radius: 12px;
  transition: all 0.1s linear;

  &:hover {
    font-size: larger;
    transition: all 0.1s linear;
  }
}
.primary {
  width: 200px;
  height: 175px;
  font-size: 40px;
  font-weight: 600;
  border: solid 5px rgb(90, 90, 90);
  border-radius: 20px;
  transition: all 0.2s ease-in-out;

  &::before {
    content: '';
    border: solid 5px rgb(90, 90, 90);
    border-radius: 10px;
    position: absolute;
    top: 0;
    left: 0;
    margin: 10px;
    right: 0;
    bottom: 0;
    transition: all 0.2s ease-in-out;
  }
  
  &:hover {
    font-size: 45px;
    transition: all 0.25s ease-in-out;
  
    &::before {
      margin: 15px;
      background-color: rgba(200, 200, 200, 0.2);
    }
  }
}