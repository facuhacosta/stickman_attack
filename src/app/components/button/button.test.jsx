import React from 'react';
import { render } from '@testing-library/react';
import { Button } from './button';

describe('Button Component', () => {

  // Matching Snapshots
  test('it should match the snapshot', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Unit tests

});