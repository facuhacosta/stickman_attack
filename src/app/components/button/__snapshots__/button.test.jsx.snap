// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component it should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="undefined"
    />
  </div>
</DocumentFragment>
`;
